src/app/court/court.component.ts

```diff
@@ -42,6 +42,11 @@ export class CourtComponent implements OnInit {
   currentAction: PageAction = PageAction.NONE;
   courtSelectorPlaceholder: String = '';
 
+  constructor(public dialog: MatDialog, private courtService: CourtsService) {
```

Comment:
I've since learned more about Angular 5 and found that it's really not best practice to be interacting with your data, especially bound data, in the constructor. Instead have your component implement `OnInit` and do this in the `ngOnInit` function